src/core/tests/TestTelemetryWriter.py

```diff
@@ -122,6 +122,11 @@ def test_write_to_new_file_if_event_file_limit_reached(self):
         telemetry_get_event_file_size_backup = self.runtime.telemetry_writer.get_file_size
         self.runtime.telemetry_writer.get_file_size = self.mock_get_file_size
 
+        # forcing wait of 1 sec to ensure new file is created, since we have mocked time.sleep in RuntimeComposer
+        init_time = time.time()
+        while time.time() < init_time + 1:
+            pass
+
```

Comment:
This spools up the CPU to 100%; since it's single-threaded, it'll only be one core but we shouldn't do this.

When the mock is created, the original is backed up right. Have that reference available and sleep using that.

Reply:
Updated - Using the original sleep method